### ISSUE TITLE
pooled.con is undefined - Cannot read property 'close' of undefined

### ISSUE THREAD

**Issue by 700software** [700software/tedious-connection-pool2#2]
https://github.com/700software/tedious-connection-pool2/blob/dbd6c998144fc4efb2bf0ff9d4b4bbd948d4bbae/lib/connection-pool.js#L230
```
C:\...\node_modules\tedious-connection-pool2\lib\connection-pool.js:230
        pooled.con.close();
                   ^

TypeError: Cannot read property 'close' of undefined
    at Timeout._onTimeout (C:\...\node_modules\tedious-connection-pool2\lib\connection-pool.js:230:20)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
```

Some edge case this can happen if there were some reconnect error - and causes app to crash. Instead app should retry indefinitely as usual.

**Comment by 700software** [comment]
I think what is happening
1. The connection is fine.
2. An error happens and it goes to `RETRY` status.
3. The idle timeout is reached and it goes to `FREE` status.

So methinks we need to make sure retry is aborted when idle timeout is reached.

**Comment by 700software** [comment]
I wonder if this got fixed in #3? Closing unless someone says otherwise.
